src/katgpucbf/fgpu/engine.py

```diff
@@ -1018,9 +1032,17 @@ async def _run_receive(self, streams: List[spead2.recv.ChunkRingStream], layout:
             in_item.n_samples = chunks[0].data.nbytes * BYTE_BITS // self.sample_bits
 
             transfer_events = []
-            for pol_data, chunk in zip(in_item.pol_data, chunks):
+            for pol, (pol_data, chunk) in enumerate(zip(in_item.pol_data, chunks)):
                 # Copy the present flags (synchronously).
                 pol_data.present[: len(chunk.present)] = chunk.present
+                # Update the digitiser saturation count (the "extra" fields holds
+                # per-heap values).
+                assert chunk.extra is not None
+                sensor = self.sensors[f"dig.pol{pol}-dig-clip-cnt"]
+                sensor.set_value(
+                    sensor.value + int(np.sum(chunk.extra, dtype=np.uint64)),
+                    timestamp=chunk.timestamp + layout.chunk_samples,
```

Comment:
Good question - any idea what MeerKAT does? I feel like it should be the digitiser (or DMC) that goes into warn rather than the F-engine though, since that's where the problem is occurring.